### PR TITLE
Delete comments

### LINKED ISSUE #72: Deleting Comments
Develop the logic to delete a comment for an article, when they click the 'trash' button

### PR DESCRIPTION
**Pull request by ebebbington**
Fixes #72 

**Description**

* Add backend logic for deleting comments
    * Checks if user is logged in and that the comment to delete is theirs

### CODE REVIEW

File: src/resources/article_comments_resource.ts [hunk 1]
```diff
@@ -99,4 +100,48 @@ export default class ArticleCommentsResource extends Drash.Http.Resource {
...
   }
+
+  public async DELETE() {
+    console.log('Handling ArticleCommentsResource DELETE.')
+    // make sure they are authorised to do so
+    const cookie = this.request.getCookie("drash_sess")
+    const user = await UserService.getLoggedInUser(cookie || "");
+    if (typeof user === "boolean") {
+      console.error("Seems like the 'user' isn\'t authenticated.")
+      this.response.status_code = 403
```
Comment (crookse) [inline src/resources/article_comments_resource.ts:111]: you can use this for this: https://github.com/drashland/deno-drash-realworld-example-app/blob/7665a7981b43a99467c7fa1440d9f173c0a707b0/src/resources/base_resource.ts#L19

File: src/resources/article_comments_resource.ts [hunk 2]
```diff
@@ -99,4 +100,48 @@ export default class ArticleCommentsResource extends Drash.Http.Resource {
...
+    // Make sure they are the author of the comment
+    const commentId = this.request.getPathParam("id")
+    console.log("going to get comments")
+    const comments = await ArticleCommentsModel.where({ author_id: user.id});
+    const isTheirComment = comments.filter(comment => {
+      return comment.id == commentId
+    }).length >= 0
+    if (!isTheirComment) {
+      console.error("User tried to delete someone elses comment")
+      this.response.status_code = 403
```
Comment (crookse) [inline src/resources/article_comments_resource.ts:128]: see comment above

File: src/vue/store/actions.js [hunk 3]
```diff
@@ -39,6 +39,7 @@ export default {
 
     console.log("User is not authenticated.");
     context.dispatch("unsetUser");
+    window.location.href = "/#/login"
```
Comment (crookse) [inline src/vue/store/actions.js:42]: router.push("/login");

Comment (ebebbington) [reply src/vue/store/actions.js:42]: THANK YOU :D I knew there was a vue way but didn't know what

File: src/resources/article_comments_resource.ts (hunk 1 above)
Comment (ebebbington) [reply src/resources/article_comments_resource.ts]: Forgot about that .. https://github.com/drashland/deno-drash-realworld-example-app/pull/75/commits/607820ea732a0243befa5d16174eb333cd2b1afb

File: src/resources/article_comments_resource.ts (hunk 2 above)
Comment (ebebbington) [reply src/resources/article_comments_resource.ts]: https://github.com/drashland/deno-drash-realworld-example-app/pull/75/commits/607820ea732a0243befa5d16174eb333cd2b1afb

File: src/vue/store/actions.js (hunk 3 above)
Comment (ebebbington) [reply src/vue/store/actions.js]: https://github.com/drashland/deno-drash-realworld-example-app/pull/75/commits/68e4dc8a743ab2ecdb872723afc9fa46cf956e6f

**Comment by ebebbington** [comment]
Btw the ci will always fail, and i can't be bothered to figure out why

**Pull request merged**